components/function-controller/design/subresources_templates_proposal.md

```diff
@@ -0,0 +1,84 @@
+# Function Subresources Templating for Serverless v1alpha2
+
+## Summery
+
+The current Serverless API allows for limited ability to configure the generated functions subresources (deployment, build job). Additionally, it doesn't allow for separate configuration for build and functions resources except for memory/cpu resources.
+
+This proposes changes to the current v1alpha2 Serverless API to support templating of function sub resources _without_ the need to rollout a new API version.
+
+## Motivation
+
+Give Serverless users the ability to:
+- Separate configuration for function deployments and build jobs.
+- Configure volume mounts for function subresources
```

Comment:
Is there an additional use case for volume mounts not only for function deployment?